### PR TITLE
Model deploy error is not informative

### PR DESCRIPTION
**Pull request by dpanshug**
Closes: [RHOAIENG-2627](https://issues.redhat.com/browse/RHOAIENG-2627)

## Description
Improved the functionality to display model error status. 
Now the user can see error message properly listed under `lastFailureInfo.message` in InferenceService CR.

image

## How Has This Been Tested?
Deploy a model with wrong AWS credentials and check it's status, error message should be proper and informative. 

## Test Impact
Added cypress test to check error message.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by dpanshug** [comment]
@lucferbux should I squash both commits together?

**Comment by lucferbux** [comment]
@dpanshug well no, I'm sorry, now that i see the commits, just keep them as they are, we might wanna have both of them split, if you are able to integrate new chagnes in the first one this is fine

**Review by lucferbux — CHANGES_REQUESTED** [review]

File: frontend/src/pages/modelServing/screens/global/utils.ts [hunk 1]
```diff
@@ -16,8 +16,8 @@ export const getInferenceServiceActiveModelState = (
   InferenceServiceModelState.UNKNOWN;
 
 export const getInferenceServiceStatusMessage = (is: InferenceServiceKind): string =>
-  is.status?.modelStatus.states.activeModelState ||
   is.status?.modelStatus.lastFailureInfo?.message ||
+  is.status?.modelStatus.states.activeModelState ||
```
[the remark this answers is not in the document]
Comment (dpanshug) [reply frontend/src/pages/modelServing/screens/global/utils.ts:20]: `targetModelState` can also return `FailedToLoad`

image


Need to check this with Model serving team. [slack thread](https://redhat-internal.slack.com/archives/C065ARTVA80/p1707923783905709)

**Review by manaswinidas — CHANGES_REQUESTED** [review]
2 things:

1. The screen keeps blinking for me when I try to hover over the failed icon, is this normal? I tried my best to capture this via screen recording

https://github.com/opendatahub-io/odh-dashboard/assets/22885912/9227b47c-4b97-4602-85e0-a66f354f3638

2. Should the error message be this long? This is overflowing.

https://github.com/opendatahub-io/odh-dashboard/assets/22885912/95bf4069-be0b-4fc5-889d-ef8f9361cc29

**Comment by dpanshug** [comment]
@manaswinidas 
> 1. The screen keeps blinking for me when I try to hover over the failed icon, is this normal? I tried my best to capture this via screen recording. You can see the blinking in the beginning.

I've noticed the blinking too and had a discussion with Lucas about it. It appears to be related to a patternfly issue with resizing large tooltips. Interestingly, when you open it in Firefox, there's no blinking. I'll take this issue up with the Patternfly team.

> 2. Should the error message be this long? This is overflowing.

I've started a discussion on [slack](https://redhat-internal.slack.com/archives/C06EJG57Z0X/p1708094127706089)

File: frontend/src/pages/modelServing/screens/global/utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dpanshug) [reply frontend/src/pages/modelServing/screens/global/utils.ts:20]: Added `targetModelState` after discussing with the model serving team ([here](https://redhat-internal.slack.com/archives/C06EJG57Z0X/p1708350552257149?thread_ts=1708094127.706089&cid=C06EJG57Z0X))

Still the user might end up on `FailedToLoad` state, as `targetModelState` can be `pending/inprogress/loaded/failedtoload`

**Review by lucferbux — APPROVED** [review]
/lgtm

File: frontend/src/pages/modelServing/screens/global/utils.ts [hunk 2]
```diff
@@ -15,10 +15,18 @@ export const getInferenceServiceActiveModelState = (
...
+  const targetModelState = is.status?.modelStatus.states.targetModelState;
+
+  return activeModelState === InferenceServiceModelState.FAILED_TO_LOAD ||
+    targetModelState === InferenceServiceModelState.FAILED_TO_LOAD
+    ? is.status?.modelStatus.lastFailureInfo?.message ??
+        activeModelState ??
+        targetModelState ??
+        'Unknown'
+    : activeModelState ?? targetModelState ?? 'Unknown';
+};
```
Comment (lucferbux) [inline frontend/src/pages/modelServing/screens/global/utils.ts:29]:
```suggestion
  const failedToLoad = InferenceServiceModelState.FAILED_TO_LOAD;
  const isFailedToLoad = activeModelState === failedToLoad || targetModelState === failedToLoad;

  const lastFailureMessage = is.status?.modelStatus.lastFailureInfo?.message;
  const stateMessage = activeModelState ?? targetModelState ?? 'Unknown';

  return isFailedToLoad ? lastFailureMessage ?? stateMessage : stateMessage;
```

Try to make the code more legible and avoid redundancy, don't hesitate to use constants to make piece of code managable inside a function, your logic is not wrong, but it takes a lot of time to read and understand what's happening, make the code easier to read.

**Review by lucferbux — CHANGES_REQUESTED** [review]
nit: Your logic is not wrong, but try to always find a legible solution smarter code is the one that can be easily read and you can get the context withing a look.

File: frontend/src/pages/modelServing/screens/global/utils.ts (hunk 2 above)
Comment (dpanshug) [reply frontend/src/pages/modelServing/screens/global/utils.ts:29]: Yes, this refactored code looks more readable. Thanks!

**Review by lucferbux — APPROVED** [review]
/lgtm

**Pull request merged**